lib/src/package_analysis/shape.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'shape.g.dart';
+
+@JsonSerializable()
+class PackageShape {
```

Comment:
Consider adding `@sealed` from `package:meta`.
```suggestion
@sealed
@JsonSerializable()
class PackageShape {
```

This is a nice way to indicate to anyone else that they should not extend or implement you subclasses of your class.
This is a data class, I hope we can all agree that no-one should ever extend it.

Reply:
I suppose the same should go for all the other Shape classes?

Reply:
Yeah, in fact it probably applies to almost all classes you create... maybe not the AnalysisContext, that one is nice that people can implement themselves, if they want to report warnings differently.